### PR TITLE
block adding same license twice

### PR DESCRIPTION
**Pull request by jgruica**
Screen Shot 2020-11-19 at 9 10 37 AM
Screen Shot 2020-11-20 at 7 56 04 AM

### CODE REVIEW

File: kotsadm/pkg/store/s3pg/license_store.go [hunk 1]
```diff
@@ -85,3 +85,19 @@ func (s S3PGStore) GetLicenseForAppVersion(appID string, sequence int64) (*kotsv
 
 	return nil, nil
 }
+
+func (s S3PGStore) CheckDoesLicenseExists(license string) (string, error) {
+	db := persistence.MustGetPGSession()
+	query := `select slug from app where license = $1`
```
Comment (divolgin) [inline kotsadm/pkg/store/s3pg/license_store.go:91]: This is not going to work if license is updated or even re-downloaded.  You have to match license ID only.

File: kotsadm/pkg/store/s3pg/license_store.go [hunk 2]
```diff
@@ -85,3 +85,58 @@ func (s S3PGStore) GetLicenseForAppVersion(appID string, sequence int64) (*kotsv
...
+			}
+			license := obj.(*kotsv1beta1.License)
+			licenses = append(licenses, license)
+		}
+	}
+
+	return licenses, nil
+}
+
+func (s S3PGStore) GetParsedLicense(licenseStr string) (*kotsv1beta1.License, error) {
```
Comment (salahalsaleh) [inline kotsadm/pkg/store/s3pg/license_store.go:134]: this  function doesn't read anything from the store, i think it should be moved outside the store

File: kotsadm/pkg/license/license.go [hunk 3]
```diff
@@ -147,3 +148,17 @@ func GetCurrentLicenseString(a *apptypes.App) (string, error) {
...
 }
+
+func CheckDoesLicenseExists(allLicenses []*kotsv1beta1.License, uploadedLicense string) (*kotsv1beta1.License, error) {
+	parsedUploadedLicense, err := store.GetStore().GetParsedLicense(uploadedLicense)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse uploaded license")
+	}
+
+	for _, license := range allLicenses {
+		if reflect.DeepEqual(license, parsedUploadedLicense) {
```
Comment (salahalsaleh) [inline kotsadm/pkg/license/license.go:159]: this should just match license id instead

File: kotsadm/pkg/license/license.go [hunk 4]
```diff
@@ -147,3 +148,17 @@ func GetCurrentLicenseString(a *apptypes.App) (string, error) {
 	}
 	return string(kotsLicense), nil
 }
+
+func CheckDoesLicenseExists(allLicenses []*kotsv1beta1.License, uploadedLicense string) (*kotsv1beta1.License, error) {
+	parsedUploadedLicense, err := store.GetStore().GetParsedLicense(uploadedLicense)
```
Comment (salahalsaleh) [inline kotsadm/pkg/license/license.go:153]: i suggest moving this function outside of the store and just use it here

File: kotsadm/pkg/store/ocistore/license_store.go [hunk 5]
```diff
@@ -39,3 +39,34 @@ func (s OCIStore) GetLicenseForAppVersion(appID string, sequence int64) (*kotsv1
 
 	return appVersion.KOTSKinds.License, nil
 }
+
+func (s OCIStore) GetAppSlugForLicense(license string) (string, error) {
+	appSlug, err := s.GetAppSlugForLicense(license)
```
Comment (salahalsaleh) [inline kotsadm/pkg/store/ocistore/license_store.go:44]: i don't understand this, why calling the same function again, and twice? won't this be stuck on a loop? you could just use this for now
```
return nil, ErrNotImplemented
```

File: kotsadm/pkg/store/ocistore/license_store.go [hunk 6]
```diff
@@ -39,3 +39,34 @@ func (s OCIStore) GetLicenseForAppVersion(appID string, sequence int64) (*kotsv1
...
+
+	if appSlug == "" {
+		return s.GetAppSlugForLicense(license)
+	}
+
+	return appSlug, nil
+}
+
+func (s OCIStore) GetAllAppLicenses() ([]*kotsv1beta1.License, error) {
+	licenses, err := s.GetAllAppLicenses()
```
Comment (salahalsaleh) [inline kotsadm/pkg/store/ocistore/license_store.go:57]: same here, just
```
return nil, ErrNotImplemented
```

File: kotsadm/pkg/store/ocistore/license_store.go [hunk 7]
```diff
@@ -39,3 +39,34 @@ func (s OCIStore) GetLicenseForAppVersion(appID string, sequence int64) (*kotsv1
...
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get license")
+	}
+
+	return licenses, nil
+}
+
+func (s OCIStore) GetParsedLicense(licenseStr string) (*kotsv1beta1.License, error) {
+	license, err := s.GetParsedLicense(licenseStr)
```
Comment (salahalsaleh) [inline kotsadm/pkg/store/ocistore/license_store.go:66]: and here
```
return nil, ErrNotImplemented
```

**Review by divolgin — APPROVED** [review]

**Review by GraysonNull — APPROVED** [review]
Front end code looks fine

**Pull request merged**